### PR TITLE
Extended fix for #9101

### LINKED ISSUE #9101: SDMolSupplier length/indexing regression in 2025.09.5 when "$$$$\n" crosses 64KiB chunk boundary
Hi RDKit team,

I found a bug in `SDMolSupplier` on `2025.09.5` that appears related to the buffered indexing change from PR #9010.

## Summary

When using:

```python
sup = Chem.SDMolSupplier(path, removeHs=False)
for i, mol in zip(range(len(sup)), sup):
    ...
```

`len(sup)` triggers index building, and if an SDF record separator `$$$$` is at the end of a 65536-byte chunk while its trailing `\n` is in the next chunk, parsing of a later molecule can fail with:

- `ERROR: Counts line too short: ''`
- `ERROR: moving to the beginning of the next molecule`

## Reproduction
`rdkit_bug_buffer.sdf` is attached (pls remove the .txt suffix).

```python
from rdkit import Chem
import rdkit

print(rdkit.__version__)  # 2025.09.5

fp = "rdkit_bug_buffer.sdf"

# A: direct streaming iteration (works)
sup = Chem.SDMolSupplier(fp, removeHs=False)
print([i for i, m in enumerate(sup) if m is None])  # []

# B: len()+iteration (fails)
sup = Chem.SDMolSupplier(fp, removeHs=False)
L = len(sup)
print(L)  # 2
print([i for i, m in zip(range(L), sup) if m is None])  # [1]
```

## Version comparison

- `2025.09.5`: reproduces
- `2025.09.4`: does not reproduce with the same file and code
- `2025.09.3`: does not reproduce with the same file and code

## Suspected root cause

In `SDMolSupplier::buildIndexTo()`, when `$$$$` is found near the end of a buffer:

```cpp
char *nlPos = match + 4;
while (nlPos < bufEnd && *nlPos != '\n') ++nlPos;
if (nlPos < bufEnd) ++nlPos;
```

If `\n` is in the next chunk, `nlPos == bufEnd`, so `++nlPos` is not executed. Then the newline start is "\n", not the real new line start. This causes downstream Mol block misalignment and the `"Counts line too short"` parse error.

[rdkit_bug_buffer.sdf.txt](https://github.com/user-attachments/files/25203459/rdkit_bug_buffer.sdf.txt)

### PR DESCRIPTION
**Pull request by RaulSofia**
#### Reference Issue
Fixes #9101 (extends what was already done)
Original fixing was done in PR #9162 
Original source of the bug was PR #9010 

#### What does this implement/fix? Explain your changes.
After version 2025.9.5, buffered reading was introduced for better performance with SDMolSupplier. This did not account for a boundary condition where a delimiter was present at the very end of the buffer. PR #9162 fixed this (somewhat) and introduced testing. However, while the test passed, similar problems still persisted on some real files I was using. The problem was that the fix was using the older `checkForEnd`, which changes the pointer position, instead of `peekCheckForEnd` (introduced in PR #9010 ) that restores the pointer position and avoids drift. The test case was not catching this because it only contained 2 molecules (drift would only be observed after that).

I changed the supplier to always use the peek version of the function, and added an extended test case with three molecules, that catches this issue.

#### Any other comments?
I did not remove the old test (although it may be redundant now, it really is just one more test).

Some other QoL changes:
- Some tests were failing on Windows, due to the "spaces in paths" issue. It is a common issue on Windows since the username folder itself usually contains a space, and i figured it wouldn't hurt to update that, it may help other people building there.
- Due to garbage collection delays (also probably only visible on Windows), a test was failing. I forced garbage collection on that one, it is just a test, doesn't hurt performance, and makes it more deterministic.
- gitignored a test file that was being generated

Ran the changes through clang too.

### CODE REVIEW

**Comment by RaulSofia** [comment]
You may have noticed that the two last commits are empty, just for retriggering the CI pipeline. First, a Java test (in DiversityPickerTests.java, test3) wasn't passing, due to randomness in the test I believe, so I relaunched. On the second attempt, some dependencies failed to download (I was seeing 404s during the build phase) and so it couldn't properly build some instances.
I finally tried a third time, no changes, and it now passes all checks

**Comment by greglandrum** [comment]
> You may have noticed that the two last commits are empty, just for retriggering the CI pipeline. First, a Java test (in DiversityPickerTests.java, test3) wasn't passing, due to randomness in the test I believe, so I relaunched. On the second attempt, some dependencies failed to download (I was seeing 404s during the build phase) and so it couldn't properly build some instances. I finally tried a third time, no changes, and it now passes all checks

Hi @RaulSofia just as an FYI for the future: you don't need to worry about doing this kind of dance to make the CI builds work. We know which tests occasionally fail and either just ignore them or trigger a re-run of the failing tests if there are any questions.

File: Code/GraphMol/FileParsers/testMolSupplier.cpp [hunk 1]
```diff
@@ -2719,3 +2719,68 @@ TEST_CASE("github9101 - $$$$ at buffer end") {
   REQUIRE(mol);
   delete mol;
 }
+
+// extends the "github9101 - $$$$ at buffer end" test case
+// just two molecules werent enough to reliably trigger the issue
+// (although the second created drift it only took effect on the next molecule
+// which would be the third). this creates the edge case but with three
+TEST_CASE("chunk boundary stream drift with 3+ molecules") {
+  std::string m1 =
```
Comment (greglandrum) [inline Code/GraphMol/FileParsers/testMolSupplier.cpp]: Please use a raw string for this so that you don't have to have all of the extra quotation marks and `\n`s

If you have questions about how to do that, take a look at the test that starts on line 2627 of this file

**Comment by RaulSofia** [comment]
@greglandrum got it

File: Projects/DbCLI/UnitTestDbCLI.py
Comment (greglandrum) [inline Projects/DbCLI/UnitTestDbCLI.py:1]: I think the changes in this file are not connected to an actual issue and don't reflect anything that is a common problem. They do create something of a mystery though.

Please remove this from this PR. If it's important to you, please create a separate PR for it and explain when it's relevant and what it does there.

**Review by greglandrum — CHANGES_REQUESTED** [review]
A couple of small changes requested

File: Code/GraphMol/FileParsers/SDMolSupplier.cpp [hunk 2]
```diff
@@ -358,11 +358,9 @@ void SDMolSupplier::buildIndexTo(unsigned int targetIdx) {
               posHold = posHold + std::streamoff(1);
               needEOL = false;
             }
-            this->checkForEnd();
```
Comment (bp-kelley) [inline Code/GraphMol/FileParsers/SDMolSupplier.cpp:361]: I think this is the right fix, I was trying to remember why it wasn't done in the first place but couldn't think of anything.

**Review by greglandrum — APPROVED** [review]
LGTM
